conda/switch_cuda_version.sh

```diff
@@ -17,11 +17,18 @@ mkdir -p /usr/local
 rm -fr /usr/local/cuda
 ln -s "$CUDA_DIR" /usr/local/cuda
 
+CUDA_VERSION_DOT=$(nvcc --version | sed -n 4p|cut -f5 -d" " | cut -f1 -d",")
+export CUDA_VERSION=${CUDA_VERSION_DOT/./}
 if [[ "$OSTYPE" == "msys" ]]; then
-    export CUDA_VERSION=`ls /usr/local/cuda/bin/cudart64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2`
-    export CUDNN_VERSION=`ls /usr/local/cuda/bin/cudnn64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2`
+    export CUDNN_VERSION=$(ls /usr/local/cuda/bin/cudnn64*.dll | head -1 | tr '._' ' ' | cut -d ' ' -f2)
+
+    # we want CUDNN_VERSION=8.1 for CUDA 11.2, not just 8
+    if [[ "$CUDA_VERSION" == '112' ]]; then
+        CUDNN_MAJOR=`grep 'define CUDNN_MAJOR' $(find /usr/local/cuda/ -name cudnn_version.h) | cut -d' ' -f3`
+        CUDNN_MINOR=`grep 'define CUDNN_MINOR' $(find /usr/local/cuda/ -name cudnn_version.h) | cut -d' ' -f3`
```

Comment:
Addressing some Shellcheck warnings:
- [SC2006](https://github.com/koalaman/shellcheck/wiki/SC2006) Use `$(...)` notation instead of legacy backticked `` `...` ``.
- [SC2046](https://github.com/koalaman/shellcheck/wiki/SC2046) Quote this to prevent word splitting
- [SC2038](https://github.com/koalaman/shellcheck/wiki/SC2038) Use -print0/-0 or find -exec + to allow for non-alphanumeric filenames.
```suggestion
        CUDNN_MAJOR=$(find /usr/local/cuda/ -name cudnn_version.h -exec grep 'define CUDNN_MAJOR' {} + | cut -d' ' -f3)
        CUDNN_MINOR=$(find /usr/local/cuda/ -name cudnn_version.h -exec grep 'define CUDNN_MINOR' {} + | cut -d' ' -f3)
```